core/src/test/scala/unit/kafka/server/ReplicaManagerTest.scala

```diff
@@ -69,7 +69,6 @@ import com.yammer.metrics.core.Gauge
 import kafka.log.remote.RemoteLogManager
 import org.apache.kafka.common.config.AbstractConfig
 import org.apache.kafka.common.internals.Topic
-import org.apache.kafka.common.message.AddPartitionsToTxnRequestData.{AddPartitionsToTxnTopic, AddPartitionsToTxnTopicCollection, AddPartitionsToTxnTransaction}
```

Comment:
For our `setUpReplicaManagerWithMockedAddPartitionsToTxnManager` we can remove the lines of code that mocks the flow for getting transaction state partitions.

    val metadataResponseTopic = Seq(new MetadataResponseTopic()
      .setName(Topic.TRANSACTION_STATE_TOPIC_NAME)
      .setPartitions(Seq(
        new MetadataResponsePartition()
          .setPartitionIndex(0)
          .setLeaderId(0)).asJava))
    transactionalTopicPartitions.foreach(tp => when(metadataCache.contains(tp)).thenReturn(true))
    when(metadataCache.getTopicMetadata(Set(Topic.TRANSACTION_STATE_TOPIC_NAME), config.interBrokerListenerName)).thenReturn(metadataResponseTopic)
    when(metadataCache.getAliveBrokerNode(0, config.interBrokerListenerName)).thenReturn(Some(node))
    when(metadataCache.getAliveBrokerNode(1, config.interBrokerListenerName)).thenReturn(None)
    
 I think the same code exists for `setupReplicaManagerWithMockedPurgatories`